component-runtime-impl/src/main/java/org/talend/sdk/component/runtime/input/InputImpl.java

```diff
@@ -48,6 +56,14 @@ public class InputImpl extends LifecycleImpl implements Input, Delegated {
 
     private transient RecordBuilderFactory recordBuilderFactory;
 
+    private transient Method checkpoint;
+
+    private transient Method shouldCheckpoint;
+
+    private transient Consumer<CheckpointState> checkpointCallback;
+
+    private boolean checkpointEnabled = Boolean.parseBoolean(System.getProperty("talend.checkpoint.enabled", "false"));
```

Comment:
Should it be final? or we may change this variable during the runtime?

Reply:
Will review it again but it was problematic to make it final for testing parts afair.

Reply:
and overloaded constructors